web/components/contract/contract-card.tsx

```diff
@@ -346,3 +395,64 @@ export function NumericResolutionOrExpectation(props: {
     </Col>
   )
 }
+
+function BetToast(props: {
+  outcome: string
+  seconds: number
+  onToastFinish: () => void
+  onToastCancel: () => void
+}) {
+  const { outcome, seconds, onToastFinish, onToastCancel } = props
+
+  // Track the number of seconds left, starting with durationMs
+  const [secondsLeft, setSecondsLeft] = useState(seconds)
+  console.log('renderings using', secondsLeft)
+
+  // Update the secondsLeft state every second
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setSecondsLeft((seconds) => seconds - 1)
+    }, 1000)
+    return () => clearInterval(interval)
+  }, [])
+
+  if (secondsLeft <= 0) {
+    console.log('finishing')
+    onToastFinish()
+    // return null
+  }
```

Comment:
Shouldn't this be in a useEffect?

e.g.
```
useEffect(() => {
  if (secondsLeft <= 0) {
    console.log('finishing')
    onToastFinish()
  }
}, [secondsLeft])
```

Reply:
Also, why not have a single setTimeout for the whole duration? You could cancel with clearTimeout on unmount.

Reply:
Oh I guess you render the seconds left...